docs/products/postgresql/concepts/upgrade-failover.rst

```diff
@@ -57,3 +57,28 @@ During maintenance updates, cloud migrations, or plan changes, the below procedu
 
 .. Note::
     The old primary server is kept alive for a short period of time (minimum 60 seconds) with a TCP forwarding setup pointing to the new primary server allowing clients to connect before learning the new IP address.
+
+Recreation of replication slots
+-------------------------------
+
+In case of failover or controlled switchover of an Aiven for PostgreSQL service, the replication slots from the old primary server are automatically recreated in the new primary server.
+
+One-node cluster
+""""""""""""""""
+
+Before replacing a node in the one-node cluster, the new node acquires information on replication slots on the original service, re-creates them, and only then the failover is performed.
+
+Multi-node cluster
+""""""""""""""""""
+
+For multi-node setups, replication slots from the primary are synchronized to the standbys periodically. At regular time intervals
+
+* Dependencies for newly-created slots are installed in the corresponding databases (currently, every 30 seconds)
+* Positions (``confirmed_flush_lsn``) of the slots are synchronized on the primary and the standbys.
+
+At the time when a failover to a standby occurs, the standby already has active replication slots with fairly up-to-date positions from the primary (with a possible 5-second delay).
+
+.. warning::
+    
+    * In case of uncontrolled failover, slots created up to 30 seconds before the failover might be lost.
+    * Arrival of recovered replication slots positions from the old primary to the new primary might be several seconds delayed. If delayed, you might receive the same entities that you've already received when you connected to the slot after the failover without specifying a position.
```

Comment:
@alexole , could you please help me improve the second sentence here `you might receive the same entities that you've already received when you connected to the slot after the failover without specifying a position.`? It's unclear what's happening here.

Reply:
When client connects to replication slots and starts reading from it, it sends back information to the server about how much it read, the server will move to position of the slot forward (`confirmed_flush_lsn`) to the position client confirmed it received data until.

When the client connects to the slot it can specify the position from which it wants to continue reading, if position is not specified it will start receiving data from `confirmed_flush_lsn` on the server.

Say, client has read until position P1, it will be stored on the server as `confirmed_flush_lsn` and if the client disconnects and connects again, the client has two options, either connect without starting position and it will receive all the data from P1 or it can specify explicitly start position P1, in normal scenario it's the same.

Now let's imaging the client has read until P1 and then fail-over happens. In this case the `confirmed_flush_lsn` on the new master might fallback to P0 (0 < P1 - P0 <~5 sec). Fail-over will trigger client re-connection and now if the client connects without a start position, it will start reading from P0 (not P1), so it will receive whatever was sent between P0 and P1 again. In order to mitigate this, it should explicitly start reading from P1.

We should also specify one important moment I forgot to mention in the beginning, in case of fail-over when there is a huge lag between the primary and standby (say master disappears), the position of replication slots will also be old, so the positions could not be bigger than the position of the standby